backend/Application/Aggregates/Contract/BackgroundServices/ContractJobsBackgroundService.cs

```diff
@@ -0,0 +1,96 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Application.Aggregates.Contract.Entities;
+using Application.Aggregates.Contract.Jobs;
+using Application.Api.GraphQL.EfCore;
+using Application.Common.FeatureFlags;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Hosting;
+
+namespace Application.Aggregates.Contract.BackgroundServices;
+
+/// <summary>
+/// Background service which executes background jobs related to Contracts.
+///
+/// When new jobs are added they should be dependency injected and added to the constructor of this service.
+/// </summary>
+internal sealed class ContractJobsBackgroundService : BackgroundService
+{
+    private readonly IContractJobFinder _jobFinder;
+    private readonly IFeatureFlags _featureFlags;
+    private readonly IDbContextFactory<GraphQlDbContext> _dbContextFactory;
+    private readonly ILogger _logger;
+
+    public ContractJobsBackgroundService(
+        IContractJobFinder jobFinder,
+        IFeatureFlags featureFlags, 
+        IDbContextFactory<GraphQlDbContext> dbContextFactory
+    )
+    {
+        _jobFinder = jobFinder;
+        _featureFlags = featureFlags;
+        _dbContextFactory = dbContextFactory;
+        _logger = Log.ForContext<ContractJobsBackgroundService>();
+    }
+    
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        if (!_featureFlags.ConcordiumNodeImportEnabled)
```

Comment:
The variable is use by all importing flows but I can see the naming is confusing.

I have now added a summary above the field to make this more clear. I keep as is to avoid breaking changes in deployment configuration files.